src/couch_proc_manager.erl

```diff
@@ -436,9 +436,8 @@ assign_proc(#client{}=Client, #proc_int{client=undefined}=Proc) ->
     assign_proc(Pid, Proc).
 
 
-return_proc(#state{} = State, #proc_int{} = ProcInt) ->
-    #proc_int{pid = Pid, lang = Lang} = ProcInt,
-    NewState = case is_process_alive(Pid) of true ->
+return_proc(#state{} = State, #proc_int{pid = Pid} = ProcInt) ->
+    case is_process_alive(Pid) of true ->
         case ProcInt#proc_int.t0 < State#state.threshold_ts of
```

Comment:
Specifically, the false clause on this case is where I think this is wrong. The threshold_ts is just for generational collection. If the proc is in the current generation then its just shoved back in the pool where it'd have to wait for an idle timeout before removing if we're over soft limit.